repose-aggregator/components/filters/uri-stripper-filter/src/test/scala/org/openrepose/filters/uristripper/UriStripperFilterTest.scala

```diff
@@ -114,6 +118,216 @@ class UriStripperFilterTest extends FunSpec with BeforeAndAfterEach with Matcher
+    }
+
+    it("should update a link by replacing the stripped token at the configured index") {
+      val config =
+        s"""<?xml version="1.0" encoding="UTF-8"?>
+           |<uri-stripper xmlns="http://docs.openrepose.org/repose/uri-stripper/v1.0" rewrite-location="false" token-index="1">
+           |    <link-resource uri-path-regex=".*">
+           |        <json token-index="2">$$.link</json>
+           |    </link-resource>
+           |</uri-stripper>
+         """.stripMargin
+
+      val respBody =
+        s"""
+           |{
+           |  "link": "http://example.com/v1/bar"
+           |}
+         """.stripMargin
+
+      filter.configurationUpdated(Marshaller.uriStripperConfigFromString(config))
+      request.setRequestURI("/v1/12345/foo")
+      setResponseBody(respBody, MimeType.APPLICATION_JSON.toString)
+
+      filter.doFilter(request, response, filterChain)
+
+      (Json.parse(response.getContentAsString) \ "link").as[String] shouldEqual "http://example.com/v1/bar/12345"
+    }
+
+    it("should not alter the body if the link cannot be located (keep)") {
+      val config =
+        s"""<?xml version="1.0" encoding="UTF-8"?>
+           |<uri-stripper xmlns="http://docs.openrepose.org/repose/uri-stripper/v1.0" rewrite-location="false" token-index="1">
+           |    <link-resource uri-path-regex=".*">
+           |        <json link-mismatch-action="keep">$$.dne</json>
+           |    </link-resource>
+           |</uri-stripper>
+         """.stripMargin
+
+      val respBody =
+        s"""
+           |{
+           |  "link": "http://example.com/v1/foo"
+           |}
+         """.stripMargin
+
+      filter.configurationUpdated(Marshaller.uriStripperConfigFromString(config))
+      request.setRequestURI("/v1/12345/foo")
+      setResponseBody(respBody, MimeType.APPLICATION_JSON.toString)
+
+      filter.doFilter(request, response, filterChain)
+
+      response.getContentAsString shouldEqual respBody
+    }
+
+    it("should remove the field if the path segment cannot be located (remove)") {
+      val config =
+        s"""<?xml version="1.0" encoding="UTF-8"?>
+           |<uri-stripper xmlns="http://docs.openrepose.org/repose/uri-stripper/v1.0" rewrite-location="false" token-index="1">
+           |    <link-resource uri-path-regex=".*">
+           |        <json link-mismatch-action="remove">$$.dne</json>
+           |    </link-resource>
+           |</uri-stripper>
+         """.stripMargin
+
+      val respBody =
+        s"""
+           |{
+           |  "link": "http://example.com/v1/foo"
+           |}
+         """.stripMargin
+
+      filter.configurationUpdated(Marshaller.uriStripperConfigFromString(config))
+      request.setRequestURI("/v1/12345/foo")
+      setResponseBody(respBody, MimeType.APPLICATION_JSON.toString)
+
+      filter.doFilter(request, response, filterChain)
+
+      response.getContentAsString shouldEqual "{}"
```

Comment:
I edited the test, but since your comment is on this line specifically, I don't think GitHub recognizes the change.

So in this case, the link should not be removed. In the case where we fail because the replacement token-index is out of bounds, it should be. There are now tests for both cases.